core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1216,7 +1205,8 @@ class DAGScheduler(
               // The epoch of the task is acceptable (i.e., the task was launched after the most
               // recent failure we're aware of for the executor), so mark the task's output as
               // available.
-              shuffleStage.addOutputLoc(smt.partitionId, status)
```

Comment:
In the old code we'd incrementally register map outputs in `ShuffleStage` and then would write the entire set of complete map outputs into `MapOutputTracker` in one shot upon stage completion (see line 1242 in the old code). Now we write this incrementally.